Src/ILGPU/Runtime/CPU/CPUMemoryBuffer.cs

```diff
@@ -72,12 +72,14 @@ public static unsafe void CPUCopyToCPU(
             ref byte sourcePtr,
             ref byte targetPtr,
             long sourceLengthInBytes,
-            long targetLengthInBytes) =>
-            Buffer.MemoryCopy(
-                Unsafe.AsPointer(ref sourcePtr),
-                Unsafe.AsPointer(ref targetPtr),
-                sourceLengthInBytes,
-                targetLengthInBytes);
+            long targetLengthInBytes)
+        {
+            ArgumentOutOfRangeException.ThrowIfLessThan(targetLengthInBytes, sourceLengthInBytes);
+            Unsafe.CopyBlock(
+                ref targetPtr,
+                ref sourcePtr,
+                (uint)sourceLengthInBytes);
```

Comment:
Cast to `uint` here will truncate the chunk of memory being copied to 4GiB. Instead, this could be done with pinning both byrefs and calling `Buffer.MemoryCopy` which does the right thing. In any case, happy to see more people discovering an amazing project that ILGPU is.